### PR TITLE
Add just syntax

### PR DESCRIPTION
**Pull request by pickfire**
cc @EricCrosson who added make support (since I removed just from it)
cc @IndianBoy42 sorry for CCing you but I think you might be interested in seeing helix using the project you worked on

I am not sure what is just default indentation so not adding one, will create an issue in just to ask.

### CODE REVIEW

**Comment by IndianBoy42** [comment]
That's cool.

1. afaik the indentation in just is flexible as long as its consistent, like always 4 spaces, always 2 spaces, always tabs, any of those is okay.
2. However my parser for just is a little janky, Specifically the 'scanner' part that handles the indentation (because that is not context free). It parses just files wrong sometimes... more than I would like. I should probably say that in my repository and file the bad cases as issues for others to work on.
3. I see you've just copied the queries into this repository, do you have some way to make sure they stay up to date?
4. Also, I don't know how filetype detection works but just works with `justfile` or `justfile` and doesnt care about capitalization. For any of the letters, so `.JuSTfIlE` is a valid file name that would be found by the `just` command. As well as conventionally files ending with `.just` can be used with `just -f`

File: runtime/queries/just/textobjects.scm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+(expression 
+    else:(expression) @conditional.inner
+)
+
+(item [(alias) (assignment) (export) (setting)]) @statement.outer
+(recipeheader) @statement.outer
+(line) @statement.outer
+
+(comment) @comment.outer
```
Comment (Omnikar) [inline runtime/queries/just/textobjects.scm:48]: Missing ending newline

**Comment by pickfire** [comment]
> I see you've just copied the queries into this repository, do you have some way to make sure they stay up to date?

We are doing that for stuff copied from nvim-treesitter as well but so far I don't think we have a solution for this. cc @the-mikedavis 

So we don't update the tree-sitter stuff on a regular basis, once we update we have to manually review and change stuff accordingly.

> Also, I don't know how filetype detection works but just works with justfile or justfile and doesnt care about capitalization. For any of the letters, so .JuSTfIlE is a valid file name that would be found by the just command. As well as conventionally files ending with .just can be used with just -f

Ah, didn't know that but I guess we just cover the normal case, at least not weird casing like the one you showed.

> Also maybe you want to support just --fmt for autoformatting?

Helix does not have support for custom formatting just yet IIRC, so far it depends on the language server.

File: runtime/queries/just/textobjects.scm (hunk 1 above)
Comment (pickfire) [reply runtime/queries/just/textobjects.scm:48]: Thanks, I missed that somehow.

**Comment by the-mikedavis** [comment]
I believe nvim-treesitter does this by having "grammar/language maintainers" own their language and update it over time / get tagged for review on changes for a language. Helix might have an advantage here because we can just copy queries from a language's `queries/` directory if they exist (although many grammar repos I've seen don't have `queries/` or highlight tests [maybe it makes sense to contribute queries upstream?]). I don't see an easy way to automate updates :/

**Comment by archseer** [comment]
Reopen if you ever fix the build issues (likely need to raise C/C++ compiler standard in helix-loader when building the grammars)

**Pull request closed without merging**